apps/src/sketchlab/SketchlabView.tsx

```diff
@@ -106,6 +108,22 @@ const SketchlabView: React.FC<LabProps<LevelProperties>> = ({
     };
   }, []);
 
+  useEffect(() => {
```

Comment:
Does this `useEffect` only fire on level change or does it fire every time sources change?

Reply:
It fires when sources change (which is debounced to max every 500ms), but only actually remounts when the level changes (or a teacher switches between students). I chatted with Sanchit and modeled it after something similar we're doing in DanceView:

https://github.com/code-dot-org/code-dot-org/blob/b11ab65ad31c9ede84e7d31802ae6be661dad348/apps/src/dance/lab2/views/DanceView.tsx#L323-L334

Reply:
Mind adding a comment explaining that it only remounts if the level changes?

Reply:
Yep can do!